### PR TITLE
Use ImageIO based OpenJPEG codec for JP2s

### PR DESCRIPTION
**Pull request by garyttierney**
More of a RFC/WIP. I've replaced the external process OpenJpegProcessor with a Java2dProcessor configured to use the [imageio-openjpeg](https://github.com/dbmdz/imageio-jnr) ImageIO plugin. This enables streaming for JP2 files and can avoid the bitmap disk copies, but isn't an ideal solution as mentioned in #563.

Some things I have to verify:
- Any uses of Metadata that was previously being constructed by Cantaloupe is compatible with the new ImageIO metadata object returned by imageio-openjpeg.
- Performance hasn't regressed for any cases

### CODE REVIEW

File: src/main/java/edu/illinois/library/cantaloupe/processor/AutomaticSelectionStrategy.java [hunk 1]
```diff
@@ -11,8 +11,8 @@
 class AutomaticSelectionStrategy implements SelectionStrategy {
 
     private static final List<Class<? extends Processor>> JP2_CANDIDATES = List.of(
-            KakaduNativeProcessor.class,
```
Comment (DiegoPino) [inline src/main/java/edu/illinois/library/cantaloupe/processor/AutomaticSelectionStrategy.java:14]: This here does not look right. KakaduNativeProcessor.class should not be removed, but instead OpenJpegProcessor.class if Java2d is the one handling it now

**Review by DiegoPino — CHANGES_REQUESTED** [review]
Can't speak for the actual code yet, but there is an error in the automatic selection strategy commit of this pull

File: src/main/java/edu/illinois/library/cantaloupe/processor/AutomaticSelectionStrategy.java (hunk 1 above)
Comment (garyttierney) [reply src/main/java/edu/illinois/library/cantaloupe/processor/AutomaticSelectionStrategy.java:14]: Oh, good catch, thanks.

**Comment by jcoyne** [comment]
@garyttierney what are your intentions here?  Are you planning on converting this to an actual pull request or can we close this draft now?

**Comment by garyttierney** [comment]
> @garyttierney what are your intentions here? Are you planning on converting this to an actual pull request or can we close this draft now?

I think we can close it for now and I'll revisit if there's a desire again to get streaming working with OpenJPEG. With the foreign function & memory interfaces stabilized in Java 22 it's now probably worth exploring a Project Panama based binding that can avoid the large amount of buffer copies the current implementations do.

**Pull request closed without merging**